Use relative path with working directory (fixes # 1515)

azhidkov opened the pull request:
Fix for this error https://github.com/go-lang-plugin-org/go-lang-idea-plugin/issues/1515

azhidkov commented:
Image with configuration window http://postimg.org/image/nl0sjv0ez/

zolotov commented:
Looks good but
1. what about completion in `file` text field (see com.goide.runconfig.GoRunUtil#installGoWithMainFileChooser)
2. it makes sense to add the same feature in test-file and test-directory run configurations (GoTestRunConfiguration)

zolotov commented on src/com/goide/runconfig/GoRunConfigurationWithMain.java:
```diff
@@ -60,7 +60,11 @@ public void writeExternal(Element element) throws WriteExternalException {
     VirtualFile file = VirtualFileManager.getInstance().findFileByUrl(VfsUtilCore.pathToUrl(getFilePath()));
     if (file == null) {
-      throw new RuntimeConfigurationError("Main file is not specified");
+      // check relative path based on working directory
```
redundant comment

zolotov commented on src/com/goide/runconfig/GoRunConfigurationWithMain.java:
```diff
@@ -60,7 +60,11 @@ public void writeExternal(Element element) throws WriteExternalException {
     if (file == null) {
-      throw new RuntimeConfigurationError("Main file is not specified");
+      // check relative path based on working directory
+      String relativePath = FileUtil.join(getWorkingDirectory(), getFilePath());
```
It's not relativePath, it's just path. Please rename.

azhidkov commented:
I added the same feature for file (http://postimg.org/image/6olizuvf1/) and directory (http://postimg.org/image/om7mvsitf/) and fixed your comments about first commit. Plus ocassionally added fix for #1758 issue. 

But didn't realize what to do with (how it should be?):
> 1. what about completion in file text field (see com.goide.runconfig.GoRunUtil#installGoWithMainFileChooser)

dlsniper commented:
I'd say please keep the fixes separated as it's easier to review them / get them merged. Thank you for your contribution ;)

zolotov commented:
> Plus ocassionally added fix for #1758 issue.

That's funny, thanks a lot :) As @dlsniper said it's simpler to merge two separate PR for these two features.

> But didn't realize what to do with (how it should be?):

Don't know yet how it should be. Let's skip this, I'll try to think something of and maybe create a separate issue.

azhidkov commented:
I forgot create separate branch for inspection, sorry :)
And fixed problem with two tests.
